pepper-apis/src/test/java/org/broadinstitute/ddp/event/pubsubtask/api/PubSubTaskCustomMessageTest.java

```diff
@@ -0,0 +1,170 @@
+import static org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil.TEST_EDUCATION;
+import static org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil.TEST_EMAIL;
+import static org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil.TEST_MARITAL_STATUS;
+import static org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil.TEST_MESSAGE_ID;
+import static org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil.TEST_PARTICIPANT_GUID;
+import static org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil.TEST_STUDY_GUID;
+import static org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil.TEST_USER_ID;
+import static org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil.buildMessage;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTaskResult.ATTR_TASK_MESSAGE_ID;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTaskResult.PubSubTaskResultType.ERROR;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTaskResult.PubSubTaskResultType.SUCCESS;
+import static org.broadinstitute.ddp.event.pubsubtask.impl.updateprofile.UpdateProfileConstants.ATTR_PARTICIPANT_GUID;
+import static org.broadinstitute.ddp.event.pubsubtask.impl.updateprofile.UpdateProfileConstants.ATTR_STUDY_GUID;
+import static org.broadinstitute.ddp.event.pubsubtask.impl.updateprofile.UpdateProfileConstants.ATTR_USER_ID;
+import static org.junit.Assert.assertEquals;
+
+import java.util.Properties;
+
+
+import com.google.gson.Gson;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil;
+import org.broadinstitute.ddp.event.pubsubtask.PubSubTaskTestUtil.TestAckReplyConsumer;
+import org.broadinstitute.ddp.util.GsonUtil;
+import org.junit.Test;
+
+public class PubSubTaskCustomMessageTest {
+
+    private final ProjectSubscriptionName projectSubscriptionName =
+            ProjectSubscriptionName.of(PROJECT_ID, PUBSUB_SUBSCRIPTION);
+    private PubSubTaskProcessorFactory testFactory;
+    private PubSubTaskReceiver pubSubTaskReceiver;
+    private PubSubTaskResultMessageCreator resultMessageCreator;
+    private PubSubTaskTestUtil.TestResultSender testResultSender;
+
+    private final Gson gson = GsonUtil.standardGson();
+
+    /**
+     * Verify that if processor registered with payloadClass=null, parse message,
+     * check that payload can be converted to a map and properties fetched from it.
+     * Create result pubsub message (successful one) out of the parsed {@link PubSubTask}
+     */
+    @Test
+    public void testCustomMessageProcessing1() {
+        init();
+        var message = buildMessage(TestProcessor1.TEST_TASK_1,
+                format("{'%s':'%s', '%s':'%s', '%s':'%s'}",
+                        EMAIL, TEST_EMAIL, EDUCATION, TEST_EDUCATION, MARITAL_STATUS, TEST_MARITAL_STATUS), true);
+        pubSubTaskReceiver.receiveMessage(message, new TestAckReplyConsumer());
```

Comment:
I think this is an ideal place to use mocks (using Mockito)!
* We can create a `var mockReply = mock(AckReplyConsumer.class)` and pass that in.
* Then we can actually verify behavior with something like `verify(mockReply, times(1)).ack()`.

Reply:
Yes, initially my idea was to mock this but then I wanted to add some test code into custom ack(), nack(), but not added..
Ok, will utilize mockito. Thanks!

Reply:
Utilized Mocks instead of custom class creation.